src/firewall/core/fw_zone.py

```diff
@@ -363,6 +363,13 @@ def add_rule_wrapper(zone, rule_str, timeout=0, sender=None):
         def remove_rule_wrapper(zone, rule_str):
             self.remove_rule(zone, Rich_Rule(rule_str=rule_str))
 
+        # These settings have no add/remove functions, they are plain attributes
```

Comment:
These are only allowed to be changed in the permanent configuration. As such we shouldn't handle them here. Instead in the client class we should remove them in `setZoneSettings()` after getting the settings with `getSettingsDbusDict()`.

This will also simplify some of your other changes. i.e. the changes is ` src/tests/dbus/zone_runtime_functional.at` aren't relevant anymore.

https://github.com/firewalld/firewalld/blob/af3f7cd074f737c584a42cf1028f18e6fa597204/doc/xml/firewalld.dbus.xml#L1320-L1333

Reply:
Ahh that makes sense. Will update the PR.

Reply:
Do you think it's okay to just have a hardcoded list of attributes to be discarded inside the `setZoneSettings` client method?

Does `UNUSED` belong in that list, or should it still be filtered out in `getSettingsDbusDict`?

Reply:
> Do you think it's okay to just have a hardcoded list of attributes to be discarded inside the `setZoneSettings` client method?

Maybe introduce getRuntimeSettingsDict() and getRuntimeSettingsDbusDict(). It has the benefit that the knowledge is localized in the FirewallClientZoneSettings class.

> Does `UNUSED` belong in that list, or should it still be filtered out in `getSettingsDbusDict`?

It should be filtered out as you've already done.